extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaConsumerMonitor.java

```diff
@@ -33,22 +31,95 @@
-                  .put("bytes-consumed-total", "kafka/consumer/bytesConsumed")
-                  .put("records-consumed-total", "kafka/consumer/recordsConsumed")
-                  .build();
-  private static final String TOPIC_TAG = "topic";
-  private static final Set<String> TOPIC_METRIC_TAGS = ImmutableSet.of("client-id", TOPIC_TAG);
+  private static final String CLIENT_ID_TAG = "client-id";
+
+  // Kafka metric name -> Kafka metric descriptor
+  private static final Map<String, KafkaConsumerMetric> METRICS =
+      Stream.of(
+          new KafkaConsumerMetric(
+              "bytes-consumed-total",
+              "kafka/consumer/bytesConsumed",
+              Set.of(CLIENT_ID_TAG, "topic"),
+              KafkaConsumerMetric.MetricType.COUNTER
+          ),
+          new KafkaConsumerMetric(
+              "records-consumed-total",
+              "kafka/consumer/recordsConsumed",
+              Set.of(CLIENT_ID_TAG, "topic"),
+              KafkaConsumerMetric.MetricType.COUNTER
+          ),
+          new KafkaConsumerMetric(
+              "fetch-total",
+              "kafka/consumer/fetch",
+              Set.of(CLIENT_ID_TAG),
+              KafkaConsumerMetric.MetricType.COUNTER
+          ),
+          new KafkaConsumerMetric(
+              "fetch-rate",
+              "kafka/consumer/fetchRate",
+              Set.of(CLIENT_ID_TAG),
+              KafkaConsumerMetric.MetricType.GAUGE
+          ),
+          new KafkaConsumerMetric(
+              "fetch-latency-avg",
+              "kafka/consumer/fetchLatencyAvg",
+              Set.of(CLIENT_ID_TAG),
+              KafkaConsumerMetric.MetricType.GAUGE
+          ),
+          new KafkaConsumerMetric(
+              "fetch-latency-max",
+              "kafka/consumer/fetchLatencyMax",
+              Set.of(CLIENT_ID_TAG),
+              KafkaConsumerMetric.MetricType.GAUGE
+          ),
+          new KafkaConsumerMetric(
+              "fetch-size-avg",
+              "kafka/consumer/fetchSizeAvg",
+              Set.of(CLIENT_ID_TAG, "topic"),
+              KafkaConsumerMetric.MetricType.GAUGE
+          ),
+          new KafkaConsumerMetric(
+              "fetch-size-max",
+              "kafka/consumer/fetchSizeMax",
+              Set.of(CLIENT_ID_TAG, "topic"),
+              KafkaConsumerMetric.MetricType.GAUGE
+          ),
+          new KafkaConsumerMetric(
+              "records-lag",
+              "kafka/consumer/recordsLag",
+              Set.of(CLIENT_ID_TAG, "topic", "partition"),
+              KafkaConsumerMetric.MetricType.GAUGE
+          ),
+          new KafkaConsumerMetric(
+              "records-per-request-avg",
+              "kafka/consumer/recordsPerRequestAvg",
+              Set.of(CLIENT_ID_TAG, "topic"),
+              KafkaConsumerMetric.MetricType.GAUGE
+          ),
+          new KafkaConsumerMetric(
+              "outgoing-byte-total",
+              "kafka/consumer/outgoingBytes",
+              Set.of(CLIENT_ID_TAG, "node-id"),
```

Comment:
We should also add constants for "topic", "partition" and "node-id".

Reply:
Updated.